OutOfSchool/OutOfSchool.WebApi/Services/RatingService.cs

```diff
@@ -209,7 +199,9 @@ public async Task<RatingDto> Update(RatingDto dto)
 
         if (await CheckRatingUpdate(dto).ConfigureAwait(false))
         {
-            var rating = await ratingRepository.Update(mapper.Map<Rating>(dto)).ConfigureAwait(false);
+            var rating = await ratingRepository.GetById(dto.Id).ConfigureAwait(false);
+            mapper.Map(dto, rating);
+            rating = await ratingRepository.Update(rating).ConfigureAwait(false);
```

Comment:
(DRY - I'm not insisting, please consider if you like it) 
I see such or similar code in many places. What it does - read from db by `dto.Id` into `entity`, take all changes from `dto` into `entity`, save it back to db. Isn't it a time to extract some helper? This looks to possible to be put on `IEntityRepositoryBase` level:

```
public async Task<TEntity> ReadAndUpdateWith<T>(TKey id, T dto, Func<T, TEntity, TEntity> map) 
{
	ArgumentNullException.ThrowIfNull(dto);

	var entity = await GetById(id).ConfigureAwait(false);
	if (entity is null)
	{
		var name = typeof(TEntity).Name;
		throw new DbUpdateConcurrencyException($"Updating failed. {name} with Id = {id} doesn't exist in the system.");
	}

	return await Update(map(dto, entity)).ConfigureAwait(false);
}
```
---
BTL, it worth mentioning from models design perspective that having twofold representation of data as `entity` and `dto` we have only `entity`-level "key" constraint (`IKeyedEntity`), but nothing for the `dto`. Thus having simple `IKeyed<TKey>` introduced and implemented by dtos could simplify beforementioned method to:
```public async Task<TEntity> ReadAndUpdateWith<T>(T dto, Func<T, TEntity, TEntity> map) where T: IKeyed<TKey>```
where you could avoid passing id as separate argument

Reply:
I added it to the base repository class. I'll change other places, where the same code exists, in the nearest future.